unstable test in the TestConcurrentLoadHistTimeout

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
--- FAIL: TestConcurrentLoadHistTimeout (1.92s)
    assertion_compare.go:313: 
        	Error Trace:	handle_hist_test.go:130
        	Error:      	"0" is not greater than "0"
        	Test:       	TestConcurrentLoadHistTimeout
        	Messages:   	[]
```

https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/37572/tests

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/statistics/handle/handle_hist_test.go
+++ b/statistics/handle/handle_hist_test.go
@@ -126,7 +126,6 @@ func TestConcurrentLoadHistTimeout(t *testing.T) {
 			break
 		}
 	}
-	stat = h.GetTableStats(tableInfo)
 	hg = stat.Columns[tableInfo.Columns[2].ID].Histogram
 	topn = stat.Columns[tableInfo.Columns[2].ID].TopN
 	require.Greater(t, hg.Len()+topn.Num(), 0)

--- a/statistics/handle/handle_hist_test.go
+++ b/statistics/handle/handle_hist_test.go
@@ -118,9 +118,11 @@ func TestConcurrentLoadHistTimeout(t *testing.T) {
 	topn = stat.Columns[tableInfo.Columns[2].ID].TopN
 	require.Equal(t, 0, hg.Len()+topn.Num())
 	// wait for timeout task to be handled
+	oldStat := stat
 	for {
 		time.Sleep(time.Millisecond * 100)
-		if len(h.StatsLoad.TimeoutColumnsCh)+len(h.StatsLoad.NeededColumnsCh) == 0 {
+		stat = h.GetTableStats(tableInfo)
+		if stat != oldStat {
 			break
 		}
 	}

